[DOCS] Add documentation for new built-in roles (#72146)
elastic/elasticsearch: x-pack/docs/en/security/authorization/built-in-roles.asciidoc

{kibana-ref}/kibana-privileges.html[Kibana privileges] for the {ml-features}.
deprecated:[7.5.0,"Replaced by <<built-in-roles-transform-user,`transform_user`>>"].

[[built-in-roles-editor]] `editor` ::
experimental:[]
Grants full access to all features in {kib} (including Solutions) and read-only access to data indices.
+
--
[NOTE]
===============================
* This role provides read access to any index that is not prefixed with a dot.

> * This role will automatically grant full access to new {kib} features as soon as they will be released.

Reviewer: ```suggestion
* This role automatically grants full access to new {kib} features as soon as they are released.
```

index templates, mappings, node and basically everything about the cluster.
However, this role does not grant permission to view the data in all indices.

[[built-in-roles-viewer]] `viewer` ::
experimental:[]
Grants read-only access to all features in {kib} (including Solutions) and to data indices.
+
--
[NOTE]
===============================
* This role provides read access to any index that is not prefixed with a dot.

> * This role will automatically grant read-only access to new {kib} features as soon as they will be available.

Reviewer: ```suggestion
* This role automatically grants read-only access to new {kib} features as soon as they are available.
```
